spec/unit/application/client_spec.rb

```diff
@@ -321,7 +321,7 @@
         expect(@pipe[0].gets).to eq("started\n")
         Process.kill("TERM", pid)
         Process.wait
-        sleep 1 # Make sure we give the converging child process enough time to finish
+        sleep 3 # Make sure we give the converging child process enough time to finish
         expect(IO.select([@pipe[0]], nil, nil, 0)).not_to be_nil
```

Comment:
can you run this in a tighter loop and then raise if it fails to turn to nil after a timeout?  (and do not bother trying to use rspec's expectation syntax, just raise an exception to fail the test).